controllers/datadogagent/merger/annotation.go

```diff
@@ -0,0 +1,31 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package merger
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// AnnotationManager is used to manage adding annotations in a PodTemplateSpec
+type AnnotationManager interface {
```

Comment:
maybe we ca create a more generic manager, like `MetadataManager` for Annotations and Labels? 
WDYT? 
can be done in another PR.